mobile/library/common/thread/posix_thread.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "envoy/thread/thread.h"
+
+#include "absl/functional/any_invocable.h"
+
+namespace Envoy {
+namespace Thread {
+
+class PosixThreadHandle {
+public:
+  explicit PosixThreadHandle(absl::AnyInvocable<void()> thread_function);
+
+  /** Returns the thread function. */
+  absl::AnyInvocable<void()>& function();
+
+  /** Returns the thread handle. */
+  pthread_t& handle();
+
+private:
+  absl::AnyInvocable<void()> thread_function_;
+  pthread_t thread_handle_;
+};
+
+/**
+ * A wrapper for POSIX thread implemented with `pthread` because `std::thread` may throw an
+ * exception and when exceptions are disabled, it will result in `abort`, which is not desirable.
+ */
+class PosixThread {
+public:
+  explicit PosixThread(PosixThreadHandle* thread_handle);
+
+  ~PosixThread();
+
+  /**
+   * Gets the thread ID. This thread ID may not be unique across different processes.
+   */
+  [[nodiscard]] ThreadId threadId();
```

Comment:
is this the POSIX thread id?  or the system thread id?  Maybe a comment with a link to the man pages would be helpful.

Reply:
Done.